functions/Set-DbaSqlLog.ps1

```diff
@@ -0,0 +1,162 @@
+                try {
+                    $currentNumLogs = $server.NumberOfLogFiles
+                    $collectionNumberLog.OriginalNumberErrorLogs = $currentNumLogs
+                }
+                catch {
+                    $collectionNumberLog.Status = "Failed collection"
+                    $collectionNumberLog
+                    Stop-Function -Message "Issue collecting current value for number of error logs" -Target $server -ErrorRecord $_ -Exception $_.Exception.InnerException.InnerException.InnerException -Continue
+                }
+
+                if ($NumberOfLog -eq $currentNumLogs) {
+                    if ($PSCmdlet.ShouldProcess($server, "Provide warning that NumberOfLog value already matches the configured value")) {
+                        Write-Message -Level Warning -Message "The provided value for NumberOfLog is already set" -Continue
+                    }
+                }
+                else {
+                    if ($PSCmdlet.ShouldProcess($server, "Setting number of logs from [$currentNumLogs] to [$NumberOfLog]")) {
+                        try {
+                            $server.NumberOfLogFiles = $NumberOfLog
+                            $server.Alter()
+                        }
+                        catch {
+                            $collectionNumberLog.Status = "Failed update"
+                            $collectionNumberLog
+                            Stop-Function -Message "Issue setting number of log files" -Target $instance -ErrorRecord $_ -Exception $_.Exception.InnerException.InnerException.InnerException -Continue
+                        }
+                    }
+                    if ($PSCmdlet.ShouldProcess($server, "Output final results of setting number of log files")) {
+                        $server.Refresh()
+                        $collectionNumberLog.CurrentNumberErrorLogs = $server.NumberOfLogFiles
+                        $collectionNumberLog
+                    }
+                }
+            }
+            if (Test-Bound 'SizeInKb') {
+                try {
+                    $currentSizeInKb = $server.ErrorInSizeKb
+                    $collectionErrorLogSize.OriginalErrorLogSizeKb = $currentSizeInKb
+                }
+                catch {
+                    $collectionErrorLogSize.Status = "Failed collection"
+                    $collectionErrorLogSize
+                    Stop-Function -Message "Issue collecting current value for number of error logs" -Target $server -ErrorRecord $_ -Exception $_.Exception.InnerException.InnerException.InnerException -Continue
+                }
+
+                if ($SizeInKb -eq $currentSizeInKb) {
+                    if ($PSCmdlet.ShouldProcess($server, "Provide warning that SizeInKb value already matches the configured value")) {
+                        Write-Message -Level Warning -Message "The provided value for SizeInKb is already set" -Continue
+                    }
```

Comment:
Looks good in -WhatIf, but makes little sense in Production use (asking for a confirm to do nothing)
Instead could use `Test-Bound 'WhatIf'` to check for WhatIf use and change messages accordingly

Reply:
Think I had some other idea writing that part but took it out or something...went and just remove the shouldprocess block.